lib/spack/spack/repo.py

```diff
@@ -1362,6 +1362,33 @@ def get_pkg_class(self, pkg_name):
         if not inspect.isclass(cls):
             tty.die("%s.%s is not a class" % (pkg_name, class_name))
 
+        new_cfg_settings = (
+            spack.config.get("packages").get(pkg_name, {}).get("package_attributes", {})
+        )
+        overidden_attrs = getattr(cls, "overidden_attrs", {})
+        for key, val in overidden_attrs.items():
+            setattr(cls, key, val)
+        # Keep track of every class attribute that is overidden by the config:
+        # if the config changes between calls to this method, we make sure to
+        # restore the original config values (in case the new config no longer
+        # sets attributes that it used to)
+        new_overidden_attrs = {}
+        for key, val in new_cfg_settings.items():
+            if hasattr(cls, key):
+                new_overidden_attrs[key] = getattr(cls, key)
+            if isinstance(val, list):
+                raise spack.config.ConfigError("Unsupported attribute value: list")
+            elif isinstance(val, dict):
+                raise spack.config.ConfigError("Unsupported attribute value: dict")
+            if key not in ["git", "url", "submodules"]:
```

Comment:
Why special handling for these?

Reply:
This relates to YAML auto-type conversion in https://github.com/spack/spack/pull/33275/files#r1012368704: since the schema anticipates these, there is no reason for the type to be inferred. Admittedly the only cases I can think of that are problematic are things like:

```
pkg_attributes:
  # I have a URL called "true", I don't want it to be inferred as a boolean
  git: "true"
```

but my personal preference is never to leave to inference what is explicitly known.

Reply:
I don't see why we need double-yaml-parsing. If we just rely on yaml parsing the first time instead of forcing everything to be a string, then we can remove this entire block.

Reply:
If you had a url "true" (weird and unlikely), you can always quote the url to turn it into a string in yaml.